### PR TITLE
[DROOLS-1139] clone EvalInvoker in EvalCondition evaluator

### PR DESCRIPTION
**Pull request by mariofusco**

### CODE REVIEW

**Comment by etirelli** [comment]
+1

File: drools-core/src/main/java/org/drools/core/rule/GroupElement.java [hunk 1]
```diff
@@ -302,7 +300,7 @@ protected GroupElement clone(boolean deepClone) {
         GroupElement cloned = new GroupElement();
         cloned.setType( this.getType() );
         for ( RuleConditionElement re : children ) {
-            cloned.addChild( deepClone && ( re instanceof GroupElement || re instanceof Pattern ) ? re.clone() : re );
+            cloned.addChild( deepClone ? re.clone() : re );
```
Comment (mariofusco) [inline drools-core/src/main/java/org/drools/core/rule/GroupElement.java:303]: @etirelli I've found another failing test case. The problem was caused that it wasn't cloning the EvalCondition here. Adding a further "|| re instanceof EvalCondition" fixed the problem, but I honestly didn't understand why it shouldn't call clone() regardless of the type of the RuleConditionElement. Any clue about this? Is this fix ok or do you see any drawback?

**Comment by mariofusco** [comment]
ok to test

**Comment by mariofusco** [comment]
Merged by https://github.com/droolsjbpm/drools/commit/8eadcaaff

**Pull request closed without merging**